### ISSUE TITLE
Does TensorFlow do some optimization (like merge a conv and BN into one conv) when converting to TFLite? If yes, is there any other layers merge optimization Tensorflow did?

### ISSUE THREAD

**Issue by jwj04ok** [tensorflow/tensorflow#47220]
<em>Please make sure that this is a feature request. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:feature_template</em>

**System information**
- TensorFlow version (you are using): 2.4.1
- Are you willing to contribute it (Yes/No):Yes

**Describe the feature and the current behavior/state.**
What kind of layers merge does Tensorflow do when converting to TFLite? (Like merge consecutive conv and BN into a new conv). I am working on some optimization part of TFLite and If Tensorflow already did some layers merge optimization part then I do not need to redo this part.

**Will this change the current api? How?**
No
**Who will benefit with this feature?**
Anyone who would like to do optimizations to TFLite
**Any Other info.**

**Comment by abattery** [comment]
In the high level of the view point, if your model is applied to the optimization and serialized with the optimized format, the TFLite model will have the same structure.

During TF -> TFL conversion, we have a sort of optimizations like dilated conversion and integrated conv + activation patterns. However, there is no optimization rule for merging conv + batchnom patterns yet.

**Comment by jwj04ok** [comment]
> In the high level of the view point, if your model is applied to the optimization and serialized with the optimized format, the TFLite model will have the same structure.
> 
> During TF -> TFL conversion, we have a sort of optimizations like dilated conversion and integrated conv + activation patterns. However, there is no optimization rule for merging conv + batchnom patterns yet.

Could you provide a list of what kind of optimizations Tensorflow already done? Therefore I could skip repeatable work.

**Comment by abattery** [comment]
@jwj04ok I am not familiar with grappler or TF optimizations. Please refer to https://www.tensorflow.org/guide/graph_optimization

**Comment by WindQAQ** [comment]
I think it's folded. (fused) batch norm is decomposed to mul, add, div and some sort of operators in `tensorflow/compiler/mlir/lite/transforms/prepare_tf.cc`, and then those ops are fused into conv in `tensorflow/compiler/mlir/lite/transforms/optimize.cc`. You can convert and validate the model

```
model = tf.keras.Sequential([
  tf.keras.layers.Conv2D(16, 3, batch_input_shape=(1, 128, 128, 3)),
  tf.keras.layers.BatchNormalization()
])
```

and check if it's folded.
